### PR TITLE
fix: fail to upvote others posts

### PR DESCRIPTION
**Pull request by huai-jie**
close https://github.com/denoland/saaskit/issues/172

### CODE REVIEW

File: utils/db.ts [hunk 1]
```diff
@@ -140,20 +147,29 @@ export async function createVote(initVote: InitVote) {
...
 
   let res = { ok: false };
   while (!res.ok) {
-    const itemByUserRes = await kv.get<Item>(itemByUserKey);
     const itemRes = await kv.get<Item>(itemKey);
     const voteByUserRes = await kv.get<Item>(voteByUserKey);
 
+    if (itemRes.value === null) throw new Error("Item does not exist");
+
+    if (itemRes.value === null) throw new Error("Item does not exist");
```
Comment (iuioiua) [inline utils/db.ts:160]:
```suggestion
```

**Review by iuioiua — APPROVED** [review]

File: routes/account/upgrade.ts [hunk 2]
```diff
@@ -28,6 +28,6 @@ export const handler: Handlers<null, AccountState> = {
       mode: "subscription",
     });
 
-    return redirect(url);
+    return redirect(url || "/");
```
Comment (iuioiua) [inline routes/account/upgrade.ts:31]: Can you please revert this change?

Comment (huai-jie) [reply routes/account/upgrade.ts]: Sure

it triggers this error though when running `deno task ok` on local machine

```
error: TS2345 [ERROR]: Argument of type 'string | null' is not assignable to parameter of type 'string'.
  Type 'null' is not assignable to type 'string'.
    return redirect(url);
                    ~~~
    at file:///....../...../saaskit/routes/account/upgrade.ts:31:21
```

**Review by iuioiua — APPROVED** [review]
LGTM! Good catch! Thank you, @huai-jie.

File: routes/account/upgrade.ts (hunk 2 above)
Comment (iuioiua) [reply routes/account/upgrade.ts]: @huai-jie, if you're still able to reproduce this issue, could you please create a GitHub issue?
